Numeric Symmetry Breaking & Warm Start

Shi-Raida opened the pull request:
Numeric Symmetry Breaking used for the CPAIOR paper with the numeric penalty placed after the template.

Warm Start used for the SOCS paper. No inter-depth warm start.

arbimo commented:
There are some XFAIL results in the "warm-up" CI job but the job is still shown as passing. Is that expected?

arbimo commented on planning/planners/src/solver.rs:
```diff
@@ -32,6 +37,33 @@ static PRINT_RAW_MODEL: EnvParam<bool> = EnvParam::new("ARIES_PRINT_RAW_MODEL",
+enum WarmUpType {
+    /// No warming up constraints
+    None,
+    /// The first plan must be exactly the same as the warm-up plan
```
Document that this causes problems with some kind of symmetry breaking?

Shi-Raida commented:
I updated the documentation of the different warm up modes.
The `causal` one does force the instances to match an action but does not force the presence of the instances. Therefore, the PSP symmetry breaking can filter some of them if they are useless.

I also refactored the `encode.rs` file to move the warm-specific logic in its own file.

The additional test files are used during the CI. The PDDL files and the plans are needed to provide the warm starting plans. I can remove the binary ones that are not used for the tests.
